include/net/net_l2.h

```diff
@@ -59,6 +59,24 @@ struct net_l2 {
 	 */
 	int (*enable)(struct net_if *iface, bool state);
 
+	/**
+	 * This function is used to override the default interface IPv6 address
+	 * lookup when sending outgoing data to dst.
+	 * Returns NULL if best_so_far value cannot be beaten.
+	 */
+	struct in6_addr *(*get_best_match_ipv6)(struct net_if *iface,
+						const struct in6_addr *addr,
+						u8_t *best_so_far);
+
+	/**
+	 * This function is used to override the default interface IPv4 address
+	 * lookup when sending outgoing data to dst.
+	 * Returns NULL if best_so_far value cannot be beaten.
+	 */
+	struct in_addr *(*get_best_match_ipv4)(struct net_if *iface,
```

Comment:
While it's usually a good idea to keep symmetry between IPv4 and IPv6, the way the addressing works in both worlds is quite different. I am not really sure we will need a way to select the source IP depending on the L2 layer in the future. If we need to improve the mechanism, it will be independent of the L2 layer.

Reply:
@jukkar @tbursztyka any thoughts on whether IPv4 best match hook is needed / wanted?  I can remove it from this patch if that's the case.